.jenkins/Jenkinsfile

```diff
@@ -11,762 +11,534 @@
+    'test': [splits: 8],
+    'test-latest': [splits: 8],
+    'test-compression': [splits: 8],
+    'stress-test': [splits: 1, size: 'small'],
+    'test-burn': [splits: 8, size: 'large'],
+    'long-test': [splits: 8],
+    'test-oa': [splits: 8],
+    'test-system-keyspace-directory': [splits: 8],
+    'jvm-dtest': [splits: 8, size: 'large'],
+    'jvm-dtest-upgrade': [splits: 8, size: 'large'],
+    'simulator-dtest': [splits: 1],
+    'dtest': [splits: 64, size: 'large'],
+    'dtest-novnode': [splits: 64, size: 'large'],
+    'dtest-latest': [splits: 64, size: 'large'],
+    'dtest-large': [splits: 8, size: 'large'],
+    'dtest-large-novnode': [splits: 8, size: 'large'],
+    'dtest-upgrade': [splits: 64, size: 'large'],
+    'dtest-upgrade-novnode': [splits: 64, size: 'large'],
+    'dtest-upgrade-large': [splits: 64, size: 'large'],
+    'dtest-upgrade-novnode-large': [splits: 64, size: 'large'],
+  ]
+  testSteps.each() {
+    it.value.put('type', 'test')
+    it.value.put('script', '.build/docker/run-tests.sh')
+    if (!it.value['size']) {
+      it.value.put('size', 'medium')
+    }
+    if (it.key.startsWith('dtest')) {
+      it.value.put('python-dtest', true)
+    }
+  }
+
+  def stepsMap = buildSteps + testSteps
+
+  // define matrix axes
+  def Map matrix_axes = [
+    arch: archsSupported(),
+    jdk: javaVersionsSupported(),
+    python: pythonsSupported(),
+    cython: ['yes', 'no'],
+    step: stepsMap.keySet(),
+    split: (1..testSteps.values().splits.max()).toList()
+  ]
+
+  def javaVersionDefault = javaVersionDefault()
+
+  def List _axes = getMatrixAxes(matrix_axes).findAll { axis ->
+    (isArchEnabled(axis['arch'])) && // skip disabled archs
+    (isJdkEnabled(axis['jdk'])) && // skip disabled jdks
+    (isStageEnabled(axis['step'])) && // skip disabled steps
+    !(axis['python'] != pythonDefault() && 'cqlsh-test' != axis['step']) && // Use only python 3.8 for all tests but cqlsh-test
+    !(axis['cython'] != 'no' && 'cqlsh-test' != axis['step']) && // cython only for cqlsh-test, disable for others
+    !(axis['jdk'] != javaVersionDefault && ('cqlsh-test' == axis['step'] || 'simulator-dtest' == axis['step'] || axis['step'].contains('dtest-upgrade'))) && // run cqlsh-test, simulator-dtest, *dtest-upgrade only with jdk11
+    // Disable splits for all but proper stages
+    !(axis['split'] > 1 && !stepsMap.findAll { entry -> entry.value.splits >= axis['split'] }.keySet().contains(axis['step'])) &&
+    // run only the build types on non-amd64
+    !(axis['arch'] != 'amd64' && stepsMap.findAll { entry -> 'build' == entry.value.type }.keySet().contains(axis['step']))
+  }
+
+  def Map tasks = [
+    // FIXME where is this used ?
+    jars: [failFast: true],
+    tests: [failFast: !isPostCommit()], // FIXME all buildSteps are always failFast
+  ]
+
+  for (def axis in _axes) {
+    def cell = axis
+    def name = getStepName(cell, stepsMap[cell.step])
+    tasks[cell.step == "jar" ? "jars" : "tests"][name] = { ->
```

Comment:
Why do we need such a mapping? Can't we use `'jars'` anywhere else in this file?

Reply:
tasks are either "jars" or "tests"

jars is used in the first stage here: https://github.com/apache/cassandra/pull/2852/files/19509dbd623af9fdbf1696694e41034b9b6b40f6#diff-917491029f94ac1ce6c114a542d2a8738ce1f2d1e866eaffe7189d2139250200R72-R74

tests are used in the second stage here: https://github.com/apache/cassandra/pull/2852/files/19509dbd623af9fdbf1696694e41034b9b6b40f6#diff-917491029f94ac1ce6c114a542d2a8738ce1f2d1e866eaffe7189d2139250200R84